ext/NFFTGPUArraysExt/implementation.jl

```diff
@@ -0,0 +1,126 @@
+mutable struct GPU_NFFTPlan{T,D, arrTc <: AbstractGPUArray{Complex{T}, D}, vecI <: AbstractGPUVector{Int32}, FP, BP, SM} <: AbstractNFFTPlan{T,D,1} 
+  N::NTuple{D,Int64}
+  NOut::NTuple{1,Int64}
+  J::Int64
+  k::Matrix{T}
+  Ñ::NTuple{D,Int64}
+  dims::UnitRange{Int64}
+  params::NFFTParams{T}
+  forwardFFT::FP
+  backwardFFT::BP
+  tmpVec::arrTc
+  tmpVecHat::arrTc
+  deconvolveIdx::vecI
+  windowLinInterp::Vector{T}
+  windowHatInvLUT::arrTc
+  B::SM
+end
+
+function AbstractNFFTs.plan_nfft(arr::Type{<:AbstractGPUArray}, k::Matrix{T}, N::NTuple{D,Int}, rest...;
+  timing::Union{Nothing,TimingStats} = nothing, kargs...) where {T,D}
+  t = @elapsed begin
+    p = GPU_NFFTPlan(arr, k, N, rest...; kargs...)
+  end
+  if timing != nothing
+    timing.pre = t
+  end
+  return p
+end
+
+function GPU_NFFTPlan(arr, k::Matrix{T}, N::NTuple{D,Int}; dims::Union{Integer,UnitRange{Int64}}=1:D,
+                 fftflags=nothing, kwargs...) where {T,D}
+
+    if dims != 1:D
+      error("GPU NFFT does not work along directions right now!")
+    end
+
+    params, N, NOut, J, Ñ, dims_ = NFFT.initParams(k, N, dims; kwargs...)
+    params.storeDeconvolutionIdx = true # GPU_NFFT only works this way
+    params.precompute = NFFT.FULL # GPU_NFFT only works this way
+
+    tmpVec = adapt(arr, zeros(Complex{T}, Ñ))
+
+    FP = plan_fft!(tmpVec, dims_)
+    BP = plan_bfft!(tmpVec, dims_)
+
+    windowLinInterp, windowPolyInterp, windowHatInvLUT, deconvolveIdx, B = NFFT.precomputation(k, N[dims_], Ñ[dims_], params)
+
+    U = params.storeDeconvolutionIdx ? N : ntuple(d->0,D)
+    tmpVecHat = adapt(arr, zeros(Complex{T}, U))
+
+    deconvIdx = adapt(arr, Int32.(deconvolveIdx))
```

Comment:
Here I changed the indices to be Int32, as these are usually more efficient for a GPU